From 9f8aedf0af8943adb43205586f2ce937d18219ec Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Wed, 13 Dec 2023 20:10:51 -0500
Subject: [PATCH] control: replace hardcoded arch name with @BUILD_ARCH@

Replace hardcoded arch name with @BUILD_ARCH@ so we can
support multiple architectures.

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 debian/control                    | 16 ++++++++--------
 debian/templates/control.extra.in | 16 ++++++++--------
 2 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/debian/control b/debian/control
index 255cd94..82796c9 100644
--- a/debian/control
+++ b/debian/control
@@ -14606,19 +14606,19 @@ Description: Linux for multiprocessor 32-bit PA-RISC (dummy package)
  This is a dummy transitional package.  It can be safely removed.
 Build-Profiles: <!stage1 !pkg.linux.nokernel !pkg.linux.nometa>
 
-Package: linux-image-stx-amd64
-Architecture: amd64
-Depends: linux-image-5.10.0-6-amd64-unsigned (= ${binary:Version}), ${misc:Depends}
+Package: linux-image-stx-@BUILD_ARCH@
+Architecture: @BUILD_ARCH@
+Depends: linux-image-5.10.0-6-@BUILD_ARCH@-unsigned (= ${binary:Version}), ${misc:Depends}
 Description: Linux for 64-bit PCs (meta-package)
  This package depends on the latest unsigned Linux kernel and modules for
  use on PCs with AMD64, Intel 64 or VIA Nano processors.
 Build-Profiles: <!stage1 !pkg.linux.nokernel>
 
-Package: linux-headers-stx-amd64
-Architecture: amd64
-Depends: linux-headers-5.10.0-6-amd64 (= ${binary:Version}), ${misc:Depends}
-Description: Header files for Linux amd64 configuration (meta-package)
+Package: linux-headers-stx-@BUILD_ARCH@
+Architecture: @BUILD_ARCH@
+Depends: linux-headers-5.10.0-6-@BUILD_ARCH@ (= ${binary:Version}), ${misc:Depends}
+Description: Header files for Linux @BUILD_ARCH@ configuration (meta-package)
  This package depends on the architecture-specific header files for the
- latest Linux kernel amd64 configuration.
+ latest Linux kernel @BUILD_ARCH@ configuration.
 Build-Profiles: <!stage1 !pkg.linux.nokernel>
 
diff --git a/debian/templates/control.extra.in b/debian/templates/control.extra.in
index c84ab8a..fde3b6f 100644
--- a/debian/templates/control.extra.in
+++ b/debian/templates/control.extra.in
@@ -43,18 +43,18 @@ Depends: linux-image-parisc, ${misc:Depends}
 Description: Linux for multiprocessor 32-bit PA-RISC (dummy package)
  This is a dummy transitional package.  It can be safely removed.
 
-Package: linux-image-stx-amd64
+Package: linux-image-stx-@BUILD_ARCH@
 Build-Profiles: <!stage1 !pkg.linux.nokernel>
-Depends: linux-image-@abiname@-amd64-unsigned (= ${binary:Version}), ${misc:Depends}
-Architecture: amd64
+Depends: linux-image-@abiname@-@BUILD_ARCH@-unsigned (= ${binary:Version}), ${misc:Depends}
+Architecture: @BUILD_ARCH@
 Description: Linux for 64-bit PCs (meta-package)
  This package depends on the latest unsigned Linux kernel and modules
  for use on PCs with AMD64, Intel 64 or VIA Nano processors.
 
-Package: linux-headers-stx-amd64
+Package: linux-headers-stx-@BUILD_ARCH@
 Build-Profiles: <!stage1 !pkg.linux.nokernel>
-Depends: linux-headers-@abiname@-amd64 (= ${binary:Version}), ${misc:Depends}
-Architecture: amd64
-Description: Header files for Linux amd64 configuration (meta-package)
+Depends: linux-headers-@abiname@-@BUILD_ARCH@ (= ${binary:Version}), ${misc:Depends}
+Architecture: @BUILD_ARCH@
+Description: Header files for Linux @BUILD_ARCH@ configuration (meta-package)
  This package depends on the architecture-specific header files for the latest
- Linux kernel amd64 configuration.
+ Linux kernel @BUILD_ARCH@ configuration.
-- 
2.30.2

